lib/rouge/lexers/json.rb

```diff
@@ -12,19 +12,55 @@ class JSON < RegexLexer
                 'application/hal+json'
 
       state :root do
-        rule /\s+/m, Text::Whitespace
-        rule /"/, Str::Double, :string
-        rule /(?:true|false|null)\b/, Keyword::Constant
-        rule /[{},:\[\]]/, Punctuation
-        rule /-?(?:0|[1-9]\d*)\.\d+(?:e[+-]?\d+)?/i, Num::Float
-        rule /-?(?:0|[1-9]\d*)(?:e[+-]?\d+)?/i, Num::Integer
+        rule /\s+/, Text::Whitespace
+        rule /{/, Punctuation, :object
+        rule /\[/, Punctuation, :array
+      end
+
+      state :object do
+        rule /\s+/, Text::Whitespace
+        rule /"/, Str::Double, :key
+        rule /:/, Punctuation, :value
+        rule /,/, Punctuation
+        rule /}/, Punctuation, :pop!
       end
 
-      state :string do
+      state :value do
+        rule /"/, Str::Char, :stringvalue
```

Comment:
Is this to differentiate between string keys and string values? It seems incorrect to tokenise this as a `Str::Char` when it's not a character class object. [Chroma's approach](https://github.com/alecthomas/chroma/blob/f1ba5bf4f3d3fd4a1bbfd8e8644cb2602b1c3881/lexers/j/json.go#L36) is to tokenise the value as a `Name::Tag`. I can't find the code in Pygments because Bitbucket is awful but, based on [their demo](http://pygments.org/demo/6820817/), they seem to do it the other way around and use the `Name::Tag` for the name (which in my opinion makes more sense). Maybe that would be better?

Reply:
Agree with the above - string values should be String tokens.